### PR TITLE
Reintroduce flight records

### PR DESCRIPTION
**Pull request by deividaspetraitis**
This PR proposes a fix for docker image build process that was failing previously ( please see #254 ).

Failure was caused by a wrong Go version since  `runtime/trace: flight recording`  package was introduced in `go1.22` as an experental feature and was not yet available at `go1.21` version, please see https://go-review.googlesource.com/c/exp/+/550257

Proposed change is to bump version from `go1.21` to `go1.22` in Docker images.
Also this PR reintroduces flight records.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,13 +1,13 @@
 # syntax=docker/dockerfile:1
 
-ARG GO_VERSION="1.21"
```
Comment (deividaspetraitis) [inline Dockerfile:3]: Proposed change: bump version from go1.21 to go1.22

File: Dockerfile.debug [hunk 2]
```diff
@@ -1,8 +1,8 @@
 # syntax=docker/dockerfile:1
 # See README.md#Debugging for more information
 
-ARG GO_VERSION="1.21"
-ARG RUNNER_IMAGE="golang:1.21"
```
Comment (deividaspetraitis) [inline Dockerfile.debug:5]: Proposed change: bump version from go1.21 to go1.22

**Review by p0mvn — APPROVED** [review]
Excellent - thank you.

I will need to update some configs in our internal repositories. I will merge this PR once that is complete.

File: CHANGELOG.md [hunk 3]
```diff
@@ -38,6 +38,7 @@ Ref: https://keepachangelog.com/en/1.0.0/
 ## Unreleased
 
 - CoinGecko pricing source support
+- Flight recording of slow requests (above 500ms) and dependency updates
```
Comment (p0mvn) [inline CHANGELOG.md]:
```suggestion
- Flight recording of slow requests and dependency updates
```

**Comment by p0mvn** [comment]
Committed a small nit to changelog.

Thank you again @deividaspetraitis

**Pull request merged**